android/src/test/java/com/instabug/reactlibrary/RNInstabugSessionReplayModuleTest.java

```diff
@@ -136,5 +135,51 @@ public void testSetUserStepsEnabled() {
         mockSessionReplay.verifyNoMoreInteractions();
     }
 
+    @Test
+
+    public void testSetSyncCallback() throws Exception  {
+        MockedStatic mockArgument = mockStatic(Arguments.class);
+        RNInstabugSessionReplayModule SRModule = spy(new RNInstabugSessionReplayModule(mock(ReactApplicationContext.class)));
+
+        CountDownLatch latch =new CountDownLatch(1);
+        SRModule.latch=latch;
+
+        when(Arguments.createMap()).thenReturn(new JavaOnlyMap());
+
+        mockSessionReplay.when(() -> SessionReplay.setSyncCallback(any(SessionSyncListener.class)))
+                .thenAnswer(new Answer<Void>() {
+                    @Override
+                    public Void answer(InvocationOnMock invocation) {
+                        ((SessionSyncListener) invocation.getArguments()[0]).onSessionReadyToSync(new SessionMetadata("device","android","1.0",20));
+                        return null;
+                    }
+                });
+
+        Thread thread= new Thread (() ->{
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            SRModule.evaluateSync(true);
+        });
```

Comment:
I don't think we should use `Thread.sleep(500)` and instead restructure the test so that it executes without hacky delays.

The idea I have in mind is we'd post `onSessionReadyToSync` on a background thread (since that's how it's expected to be called by the Android SDK), call `setSyncCallback`, when the `IBG_SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION` event is posted we call `evaluateSync` and after that we assert on the results.

What do you think?